### ISSUE TITLE
Calendar is missing focus trap for Screen Reader users (<p-datepicker>)

### ISSUE THREAD

**Issue by KrishnanAmbuli5501** [primefaces/primeng#18498]
Focus should remain trapped inside the calendar until a date is selected.
we cannot able to use p-datepicker without using mouse.

Example:
https://www.w3.org/WAI/ARIA/apg/patterns/dialog-modal/examples/datepicker-dialog/
